core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -124,28 +134,28 @@ public OperationResponse executeOperation(Operation operation, OperationMessageH
         return OperationResponse.Factory.createSimple(execute(operation));
     }
 
-    /**
-     * <b>Not supported!</b>
-     */
     @Override
     public AsyncFuture<ModelNode> executeAsync(ModelNode modelNode, OperationMessageHandler handler) {
-        throw new UnsupportedOperationException("Asynchronous execution is not supported by " + getClass().getName());
+        ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
When is the executor shut down? I guess the `...AsyncFutureTask` doesn't shutdown the executor after it's completed, so you'll keep leaking threads.

Given that this class has a `close` method, maybe it would be better to have one `Executors.newCachedThreadPool()` executor for the entire instance of this class, which you could shutdown in the `close` method?

Reply:
Thanks for advice. Calling `shutdown()` was missing. Use only one `ExecutorService` for entire instance is better solution. I fixed it in suggested way.